MSAL/src/native_auth/cache/MSALNativeAuthTokens.swift

```diff
@@ -0,0 +1,37 @@
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+@_implementationOnly import MSAL_Private
+
+class MSALNativeAuthTokens {
```

Comment:
will we return this class in public api or it is for private logic only? If it is only for private, why don't we use MSIDTokenResult?

Reply:
This object is not going to be returned externally, but the idea here was to reduce its scope. We just needed these 3 properties (`accessToken`, `refreshToken`, `rawIdToken`); using `MSIDTokenResult` we would also need `account`, `authority` and `correlationId` since they are not nullable.

Reply:
It's for public. That `@_implementationOnly import MSAL_Private` is a system we use inside the framework to access Objective C code without exposing it to the Public.
Because `MSALNativeAuthTokens` is part of `MSALNativeAuthUserAccountResult` which is public, we didn't want `MSIDTokenResult` to also be public so we only took parts of it, the 3 tokens mainly.
Let me know if that clears it up